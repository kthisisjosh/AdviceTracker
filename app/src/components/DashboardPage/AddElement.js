import React from "react"
import { Editor } from "@tinymce/tinymce-react"
import { Grid, Paper, Button, Tooltip } from "@material-ui/core"
import Scroll from "react-scroll"
import AddIcon from "@material-ui/icons/Add"

var Element = Scroll.Element

const AddElement = (props) => {
    return (
        <Element style={{ margin: "2vh" }}>
            <Paper style={{ width: "100%" }}>
                <Grid item style={{ marginLeft: "1.5vw", marginRight: "1.5vw", paddingTop: "0.75vh" }}>
                    <Editor
                        apiKey={process.env.REACT_APP_TINY_API_KEY}
                        initialValue="<p>Add your advice here!</p>"
                        init={{
                            height: 150,
                            menubar: false,
                            plugins: [
                                "advlist autolink lists link image charmap print preview anchor",
                                "searchreplace visualblocks code fullscreen",
                                "insertdatetime media table paste code help wordcount",
                            ],
                            toolbar:
                                "undo redo | formatselect | bold italic backcolor | \
             alignleft aligncenter alignright alignjustify | \
             bullist numlist outdent indent | removeformat | help",
                        }}
                        onEditorChange={props.handleEditorChange}
                    />
                </Grid>
                <Grid item style={{ marginBottom: "1vh", marginLeft: "1.5vw", paddingTop: "1vh", paddingBottom: "1vh" }}>
                    <Tooltip title="Add to Inbox">
                        <Button
                            size="small"
                            variant="contained"
                            onClick={props.handleAddClick}
                            style={{ backgroundColor: "F2994A" }}
                            startIcon={<AddIcon />}
                        >
                            Add
                        </Button>
                    </Tooltip>
                </Grid>
            </Paper>
            <Grid item></Grid>
        </Element>
    )
}

export default AddElement
