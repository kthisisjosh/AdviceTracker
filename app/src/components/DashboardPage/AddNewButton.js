import React from "react"
import { Button } from "@material-ui/core"

const AddNewButton = () => {
    return (
        <Button variant="contained" style={{ backgroundColor: "F2994A" }}>
            + Add New
        </Button>
    )
}

export default AddNewButton
