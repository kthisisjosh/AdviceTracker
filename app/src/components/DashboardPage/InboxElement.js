import React from "react"
import { Grid, Paper, Typography, IconButton, Tooltip } from "@material-ui/core"
import Scroll from "react-scroll"
import DeleteIcon from "@material-ui/icons/Delete"
import LibraryAddIcon from "@material-ui/icons/LibraryAdd"

var Element = Scroll.Element

const InboxElement = (props) => {
    return (
        <Element style={{ margin: "2vh", height: "auto" }}>
            <Paper style={{ width: "100%" }}>
                <Grid item style={{ marginLeft: "1.5vw", marginRight: "1.5vw", paddingTop: "1.5vh" }}>
                    <Typography variant="body1" dangerouslySetInnerHTML={{__html: props.advice.content}} />
                </Grid>
                <Grid item style={{ marginBottom: "1vh", marginLeft: "1vw" }}>
                    <Tooltip title="Add to category">
                        <IconButton onClick={props.handleAdd} aria-label="add to category">
                            <LibraryAddIcon />
                        </IconButton>
                    </Tooltip>
                    <Tooltip title="Delete">
                        <IconButton onClick={props.handleDelete} aria-label="delete">
                            <DeleteIcon />
                        </IconButton>
                    </Tooltip>
                </Grid>
            </Paper>
            <Grid item></Grid>
        </Element>
    )
}

export default InboxElement
