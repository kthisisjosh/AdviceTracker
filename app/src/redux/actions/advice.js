import {
    GET_INBOX_ADVICE,
    SUBMIT_INBOX_ADVICE,
    DELETE_INBOX_ADVICE,
    DELETE_CATEGORY,
    DELETE_SUBCATEGORY,
    GET_ADVICE,
    SUBMIT_CATEGORY,
    SUBMIT_ADVICE,
    SUBMIT_SUBCATEGORY,
} from "../types/advice"
import { v4 as uuidv4 } from "uuid"

export const getInboxAdvice = (id) => async (dispatch) => {
    try {
        const url = "https://advicetracker.life/api/advice/inbox/" + id
        const token = localStorage.getItem("jwtToken")
        await fetch(url, {
            method: "GET",
            headers: { "Content-Type": "application/json", Authorization: `Bearer ${token}` },
        })
            .then((response) => response.json())
            .then((data) => {
                dispatch({ type: GET_INBOX_ADVICE, payload: data })
            })
    } catch (error) {
        console.log(error)
    }
}

export const getAdvice = (id) => async (dispatch) => {
    try {
        const url = "https://advicetracker.life/api/advice/" + id
        const token = localStorage.getItem("jwtToken")
        await fetch(url, {
            method: "GET",
            headers: { "Content-Type": "application/json", Authorization: `Bearer ${token}` },
        })
            .then((response) => response.json())
            .then((data) => {
                dispatch({ type: GET_ADVICE, payload: data })
            })
    } catch (error) {
        console.log(error)
    }
}

export const submitInboxAdvice = (advice, id) => async (dispatch) => {
    try {
        const url = "https://advicetracker.life/api/advice/inbox/"
        const token = localStorage.getItem("jwtToken")
        const newInboxAdvice = {
            content: advice,
            inInbox: 1,
            userID: id,
            likes: null,
            datePosted: null,
            comments: [],
            adviceID: uuidv4(),
        }
        await fetch(url, {
            method: "POST",
            headers: {
                Authorization: `Bearer ${token}`,
                "Content-Type": "application/json",
            },
            body: JSON.stringify(newInboxAdvice),
        }).then(() => {
            dispatch({ type: SUBMIT_INBOX_ADVICE, payload: newInboxAdvice })
        })
    } catch (error) {
        console.log(error)
    }
}

export const submitAdvice = (advice, category, currSubcategory, id) => async (dispatch) => {
    try {
        const url = "https://advicetracker.life/api/advice/"
        const token = localStorage.getItem("jwtToken")
        const newAdvice = {
            content: advice,
            inInbox: 0,
            subcategoryID: currSubcategory.subcategoryID,
            userID: id,
            likes: null,
            datePosted: null,
            comments: [],
            adviceID: uuidv4(),
        }
        const filteredSubcategories = category.subcategories.filter((subcategory) => {
            return subcategory.subcategoryID !== currSubcategory.subcategoryID
        })

        const newSubcategory = { ...currSubcategory, advice: [newAdvice, ...currSubcategory.advice] }

        const updatedCategory = { ...category, subcategories: [newSubcategory, ...filteredSubcategories] }

        dispatch({ type: SUBMIT_ADVICE, payload: updatedCategory })

        await fetch(url, {
            method: "POST",
            headers: {
                Authorization: `Bearer ${token}`,
                "Content-Type": "application/json",
            },
            body: JSON.stringify(newAdvice),
        })
    } catch (error) {
        console.log(error)
    }
}

export const submitCategory = (category, id) => async (dispatch) => {
    try {
        const url = "https://advicetracker.life/api/advice/categories/"
        const token = localStorage.getItem("jwtToken")
        const updatedCategory = {
            name: category.title,
            categoryID: uuidv4(),
            userID: id,
            description: category.description,
            isSubcategory: 0,
            subcategoryID: null,
        }
        await fetch(url, {
            method: "POST",
            headers: {
                Authorization: `Bearer ${token}`,
                "Content-Type": "application/json",
            },
            body: JSON.stringify(updatedCategory),
        }).then(() => {
            dispatch({ type: SUBMIT_CATEGORY, payload: { ...updatedCategory, subcategories: [] } })
        })
    } catch (error) {
        console.log(error)
    }
}

export const submitSubCategory = (name, category, id) => async (dispatch) => {
    try {
        const url = "https://advicetracker.life/api/advice/subcategories/"
        const token = localStorage.getItem("jwtToken")
        const newSubCategory = {
            name: name,
            categoryID: category.categoryID,
            userID: id,
            description: null,
            isSubcategory: 1,
            subcategoryID: uuidv4(),
            advice: [],
        }

        const updatedCategory = { ...category, subcategories: [newSubCategory, ...category.subcategories] }

        dispatch({ type: SUBMIT_SUBCATEGORY, payload: updatedCategory })
        await fetch(url, {
            method: "POST",
            headers: {
                Authorization: `Bearer ${token}`,
                "Content-Type": "application/json",
            },
            body: JSON.stringify(newSubCategory),
        })
    } catch (error) {
        console.log(error)
    }
}

export const deleteInboxAdvice = (advice) => async (dispatch) => {
    try {
        dispatch({ type: DELETE_INBOX_ADVICE, payload: advice.adviceID })
        const token = localStorage.getItem("jwtToken")
        const url = "https://advicetracker.life/api/advice/inbox/" + advice.adviceID
        await fetch(url, {
            method: "DELETE",
            headers: {
                Authorization: `Bearer ${token}`,
                "Content-Type": "application/json",
            },
        })
    } catch (error) {
        console.log(error)
    }
}

export const deleteCategory = (id) => async (dispatch) => {
    try {
        dispatch({ type: DELETE_CATEGORY, payload: id })
        const token = localStorage.getItem("jwtToken")
        const url = "https://advicetracker.life/api/advice/categories/" + id
        await fetch(url, {
            method: "DELETE",
            headers: {
                Authorization: `Bearer ${token}`,
                "Content-Type": "application/json",
            },
        })
    } catch (error) {
        console.log(error)
    }
}

export const deleteSubCategory = (category, id) => async (dispatch) => {
    try {
        const filteredSubcategories = category.subcategories.filter((subcategory) => subcategory.subcategoryID !== id)
        dispatch({ type: DELETE_SUBCATEGORY, payload: { ...category, subcategories: [filteredSubcategories] } })
        const token = localStorage.getItem("jwtToken")
        const url = "https://advicetracker.life/api/advice/subcategories/" + id
        await fetch(url, {
            method: "DELETE",
            headers: {
                Authorization: `Bearer ${token}`,
                "Content-Type": "application/json",
            },
        })
    } catch (error) {
        console.log(error)
    }
}
