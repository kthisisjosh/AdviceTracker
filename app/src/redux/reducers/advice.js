import {
    GET_INBOX_ADVICE,
    SUBMIT_INBOX_ADVICE,
    DELETE_INBOX_ADVICE,
    DELETE_CATEGORY,
    DELETE_SUBCATEGORY,
    GET_ADVICE,
    SUBMIT_CATEGORY,
    SUBMIT_ADVICE,
    SUBMIT_SUBCATEGORY,
} from "../types/advice"

const initialState = {
    inboxAdvice: [],
    categories: [],
    browseAdvice: [],
}

export default (state = initialState, action) => {
    switch (action.type) {
        case GET_INBOX_ADVICE:
            return { ...state, inboxAdvice: [...action.payload] }

        case SUBMIT_INBOX_ADVICE:
            return {
                ...state,
                inboxAdvice: [action.payload, ...state.inboxAdvice],
            }
        case DELETE_INBOX_ADVICE:
            return {
                ...state,
                inboxAdvice: state.inboxAdvice.filter((advice) => advice.adviceID !== action.payload),
            }
        case DELETE_CATEGORY:
            return {
                ...state,
                categories: state.categories.filter((category) => category.categoryID !== action.payload),
            }
        case DELETE_SUBCATEGORY:
            return {
                ...state,
                categories: [...state.categories, action.payload],
            }
        case GET_ADVICE:
            return {
                ...state,
                categories: [...action.payload],
            }
        case SUBMIT_CATEGORY:
            return {
                ...state,
                categories: [action.payload, ...state.categories],
            }
        case SUBMIT_SUBCATEGORY:
            const filteredCategories = state.categories.filter((category) => {
                return category.categoryID !== action.payload.categoryID
            })
            return {
                ...state,
                categories: [action.payload, ...filteredCategories],
            }
        case SUBMIT_ADVICE:
            const filteredCategories2 = state.categories.filter((category) => {
                return category.categoryID !== action.payload.categoryID
            })
            return {
                ...state,
                categories: [action.payload, ...filteredCategories2],
            }
        default:
            return state
    }
}
