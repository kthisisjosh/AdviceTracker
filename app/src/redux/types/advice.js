export const GET_INBOX_ADVICE = "GET_INBOX_ADVICE"
export const SUBMIT_INBOX_ADVICE = "SUBMIT_INBOX_ADVICE"
export const DELETE_INBOX_ADVICE = "DELETE_INBOX_ADVICE"
export const GET_ADVICE = "GET_ADVICE"
export const SUBMIT_CATEGORY = "SUBMIT_CATEGORY"
