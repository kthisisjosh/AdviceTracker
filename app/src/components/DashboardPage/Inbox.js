import React, { Fragment, useState } from "react"
import { Grid, Typography } from "@material-ui/core"
import AddNewButton from "./AddNewButton"
import InboxScrollPane from "./InboxScrollPane"

const Inbox = (props) => {

    return (
        <Fragment>
            <Grid container style={{ marginTop: "3vh", marginRight: "15vw", marginBottom: "3vh", marginLeft: "15vw", width: "auto", height: "auto"}}>
                <Grid container>
                    <Typography variant="h4" align="left" style={{ marginRight: "2.5vw" }}>
                        Inbox
                    </Typography>
                    <AddNewButton handleAddClick={props.handleAddClick}/>
                </Grid>

                <Grid container style={{width: "100%", height: "15vw"}}>
                    <InboxScrollPane handleSubmit={props.handleSubmit} handleEditorChange={props.handleEditorChange} toAdd={props.toAdd} inbox={props.inbox}/>
                </Grid>
            </Grid>
        </Fragment>
    )
}

export default Inbox
