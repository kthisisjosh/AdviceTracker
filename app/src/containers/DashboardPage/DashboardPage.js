import React, { Fragment, useEffect, useState } from "react"
import Header from "../Header/Header"
import Navbar from "../../components/Navbar/Navbar"
import Footer from "../../components/Footer/Footer"
import Inbox from "../../components/DashboardPage/Inbox"
import { connect } from "react-redux"

import { getInboxAdvice, submitInboxAdvice, deleteInboxAdvice} from "../../redux/actions/advice"
import { useHistory } from "react-router-dom"

const DashboardPage = (props) => {
    const { getInboxAdvice, deleteInboxAdvice, inboxAdvice, submitInboxAdvice, isAuthenticated, user } = props
    const history = useHistory()
    const [toAdd, setToAdd] = useState(false)
    const [submitAdvice, setSubmitAdvice] = useState("")

    useEffect(() => {
        if (!inboxAdvice.length && isAuthenticated) {
            getInboxAdvice(user.userID)
        } else if (!isAuthenticated) {
            history.push("/login")
        }
    }, [getInboxAdvice, history, inboxAdvice.length, isAuthenticated, user.userID])

    const handleAddClick = () => {
        setToAdd(true)
    }

    const handleSubmit = () => {
        submitInboxAdvice(submitAdvice, user.userID)
        setSubmitAdvice("")
        setToAdd(false)
    }

    const handleEditorChange = (content, editor) => {
        setSubmitAdvice(content)
    }

    const handleAddToCategory = (event) => {
        console.log("add")
        console.log(event)
    }

    const handleDelete = (advice) => {
        deleteInboxAdvice(advice)
    }

    return (
        <Fragment>
            <Header />
            <Navbar />
            {isAuthenticated && (
                <Inbox
                    inbox={inboxAdvice}
                    handleAddToCategory={handleAddToCategory}
                    handleDelete={handleDelete}
                    handleAddClick={handleAddClick}
                    handleSubmit={handleSubmit}
                    handleEditorChange={handleEditorChange}
                    toAdd={toAdd}
                />
            )}
            <Footer />
        </Fragment>
    )
}

const mapStateToProps = ({ adviceState, authState }) => ({
    inboxAdvice: adviceState.inboxAdvice,
    isAuthenticated: authState.isAuthenticated,
    user: authState.user,
})

const mapDispatchToProps = { getInboxAdvice, submitInboxAdvice, deleteInboxAdvice }

export default connect(mapStateToProps, mapDispatchToProps)(DashboardPage)
