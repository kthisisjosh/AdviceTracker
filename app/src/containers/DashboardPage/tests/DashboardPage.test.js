import React from "react"
import DashboardPage from "../DashboardPage"
import { BrowserRouter as Router } from "react-router-dom"
import renderer from "react-test-renderer"
import store from "../../../redux"
import { Provider } from "react-redux"
import adviceReducer from "../../../redux/reducers/advice"
import { GET_INBOX_ADVICE } from "../../../redux/types/advice"

describe("<DashboardPage />", () => {
    it("should render and match the snapshot", () => {
        const component = renderer.create(
            <Provider store={store}>
                <Router>
                    <DashboardPage />
                </Router>
            </Provider>
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    describe("DashboardPage Inbox Advice Reducer", () => {
        let state
        beforeEach(() => {
            state = {
                inboxAdvice: [],
                categories: [],
                browseAdvice: [],
            }
        })

        it("should return the initial state", () => {
            const expectedResult = state
            expect(adviceReducer(undefined, {})).toEqual(expectedResult)
        })

        it("should dispatch GET_INBOX_ADVICE correctly", () => {
            const fixture = [
                {
                    adviceID: "test",
                    userID: "test",
                    inInbox: 1,
                    content: "test",
                    category: null,
                    numOfLikes: null,
                    datePosted: null,
                },
            ]

            const expectedResult = {
                inboxAdvice: [
                    {
                        adviceID: "test",
                        userID: "test",
                        inInbox: 1,
                        content: "test",
                        category: null,
                        numOfLikes: null,
                        datePosted: null,
                    },
                ],
                categories: [],
                browseAdvice: [],
            }

            expect(adviceReducer(state, { type: GET_INBOX_ADVICE, payload: fixture })).toEqual(expectedResult)
        })
    })
})
