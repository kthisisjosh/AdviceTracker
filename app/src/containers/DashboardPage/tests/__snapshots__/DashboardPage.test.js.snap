// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardPage /> should render and match the snapshot 1`] = `
Array [
  <header
    id="header"
    style={
      Object {
        "backgroundColor": "#434343",
        "margin": "0",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
      style={
        Object {
          "alignContent": "center",
          "height": "60px",
        }
      }
    >
      <div
        className="MuiGrid-root landing-logo MuiGrid-item"
        style={
          Object {
            "marginLeft": "14vw",
            "marginTop": "1vh",
          }
        }
      >
        <a
          href="/"
        >
          <img
            alt="logo"
            src="https://advicetracker.life/AdviceLogoLight.png"
            style={
              Object {
                "transform": "none",
              }
            }
          />
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
        style={
          Object {
            "marginLeft": "auto",
            "marginRight": "5vw",
          }
        }
      >
        <div
          className="MuiGrid-root toggle-grid MuiGrid-item MuiGrid-grid-md-6"
        >
          <div
            className="react-toggle react-toggle-small"
            onClick={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="react-toggle-track"
            >
              <div
                className="react-toggle-track-check"
              >
                <img
                  alt="moon"
                  aria-hidden={true}
                  height="14"
                  src="./moon.png"
                  style={
                    Object {
                      "pointerEvents": "none",
                    }
                  }
                  width="16"
                />
              </div>
              <div
                className="react-toggle-track-x"
              >
                <img
                  alt="sun"
                  aria-hidden={true}
                  height="14"
                  src="./sun.png"
                  style={
                    Object {
                      "pointerEvents": "none",
                    }
                  }
                  width="16"
                />
              </div>
            </div>
            <div
              className="react-toggle-thumb"
            />
            <input
              className="react-toggle-screenreader-only"
              defaultChecked={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="checkbox"
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
        >
          <div>
            <div
              className="bm-menu-wrap"
              id=""
              style={
                Object {
                  "MozTransform": "translate3d(100%, 0, 0)",
                  "MsTransform": "translate3d(100%, 0, 0)",
                  "OTransform": "translate3d(100%, 0, 0)",
                  "WebkitTransform": "translate3d(100%, 0, 0)",
                  "height": "100%",
                  "position": "fixed",
                  "right": 0,
                  "transform": "translate3d(100%, 0, 0)",
                  "transition": "all 0.5s",
                  "width": "300px",
                  "zIndex": 1100,
                }
              }
            >
              <div
                className="bm-menu"
                style={
                  Object {
                    "boxSizing": "border-box",
                    "height": "100%",
                    "overflow": "auto",
                  }
                }
              >
                <nav
                  className="bm-item-list"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text bm-item landing-header-link"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "display": "block",
                        "margin": "0",
                        "textDecoration": "none",
                      }
                    }
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <span
                        className="MuiTypography-root MuiTypography-button MuiTypography-alignLeft"
                        style={
                          Object {
                            "textDecoration": "none",
                          }
                        }
                      >
                        Login
                      </span>
                    </span>
                  </button>
                </nav>
              </div>
              <div
                style={Object {}}
              >
                <div
                  className="bm-cross-button"
                  style={
                    Object {
                      "height": 24,
                      "position": "absolute",
                      "right": 8,
                      "top": 8,
                      "width": 24,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "position": "absolute",
                        "right": "14px",
                        "top": "6px",
                      }
                    }
                  >
                    <span
                      className="bm-cross"
                      style={
                        Object {
                          "height": 14,
                          "position": "absolute",
                          "transform": "rotate(45deg)",
                          "width": 3,
                        }
                      }
                    />
                    <span
                      className="bm-cross"
                      style={
                        Object {
                          "height": 14,
                          "position": "absolute",
                          "transform": "rotate(-45deg)",
                          "width": 3,
                        }
                      }
                    />
                  </span>
                  <button
                    onClick={[Function]}
                    style={
                      Object {
                        "background": "transparent",
                        "border": "none",
                        "cursor": "pointer",
                        "fontSize": 0,
                        "height": "100%",
                        "left": 0,
                        "margin": 0,
                        "padding": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    Close Menu
                  </button>
                </div>
              </div>
            </div>
            <div
              style={Object {}}
            >
              <div
                className="bm-burger-button"
                style={
                  Object {
                    "zIndex": 1000,
                  }
                }
              >
                <span>
                  <span
                    className="bm-burger-bars"
                    style={
                      Object {
                        "height": "20%",
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": "0%",
                      }
                    }
                  />
                  <span
                    className="bm-burger-bars"
                    style={
                      Object {
                        "height": "20%",
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": "40%",
                      }
                    }
                  />
                  <span
                    className="bm-burger-bars"
                    style={
                      Object {
                        "height": "20%",
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": "80%",
                      }
                    }
                  />
                </span>
                <button
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "background": "transparent",
                      "border": "none",
                      "cursor": "pointer",
                      "fontSize": 0,
                      "height": "100%",
                      "left": 0,
                      "margin": 0,
                      "padding": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                >
                  Open Menu
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root navbar-grid MuiGrid-item"
        style={
          Object {
            "marginLeft": "auto",
            "marginRight": "15vw",
            "textAlign": "center",
          }
        }
      >
        <div
          className="react-toggle"
          onClick={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="react-toggle-track"
          >
            <div
              className="react-toggle-track-check"
            >
              <img
                alt="moon"
                aria-hidden={true}
                height="14"
                src="https://advicetracker.life/moon.png"
                style={
                  Object {
                    "pointerEvents": "none",
                  }
                }
                width="16"
              />
            </div>
            <div
              className="react-toggle-track-x"
            >
              <img
                alt="sun"
                aria-hidden={true}
                height="14"
                src="https://advicetracker.life/sun.png"
                style={
                  Object {
                    "pointerEvents": "none",
                  }
                }
                width="16"
              />
            </div>
          </div>
          <div
            className="react-toggle-thumb"
          />
          <input
            className="react-toggle-screenreader-only"
            defaultChecked={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="checkbox"
          />
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text landing-header-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "margin": "0 0 0 5vw",
              "textDecoration": "none",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              Login
            </span>
          </span>
        </button>
      </div>
    </div>
  </header>,
  <nav
    className="navbar"
    style={
      Object {
        "backgroundColor": "#EBEBEB",
        "margin": "0",
        "padding": "2vh 10vw 2vh 10vw",
      }
    }
  >
    <a
      className="landing-header-link"
      href="/dashboard"
      onClick={[Function]}
      style={
        Object {
          "margin": "0 5vw 0 5vw",
          "textDecoration": "none",
        }
      }
    >
      <span
        className="MuiTypography-root MuiTypography-button"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        dashboard
      </span>
    </a>
    <a
      className="landing-header-link"
      href="/browse"
      onClick={[Function]}
      style={
        Object {
          "margin": "0 5vw 0 5vw",
          "textDecoration": "none",
        }
      }
    >
      <span
        className="MuiTypography-root MuiTypography-button"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        browse
      </span>
    </a>
    <a
      className="landing-header-link"
      href="/submit"
      onClick={[Function]}
      style={
        Object {
          "margin": "0 5vw 0 5vw",
          "textDecoration": "none",
        }
      }
    >
      <span
        className="MuiTypography-root MuiTypography-button"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        submit
      </span>
    </a>
  </nav>,
  <div
    className="MuiGrid-root inbox-grid MuiGrid-container"
    style={
      Object {
        "height": "auto",
        "marginBottom": "3vh",
        "marginLeft": "15vw",
        "marginRight": "15vw",
        "marginTop": "2vh",
        "width": "auto",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft"
        style={
          Object {
            "marginRight": "2.5vw",
          }
        }
      >
        Inbox
      </h4>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "F2994A",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          Add New
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container"
      style={
        Object {
          "height": "auto",
          "width": "100%",
        }
      }
    >
      <div
        className="element"
        id="containerElement"
        style={
          Object {
            "height": "auto",
            "marginTop": "1vh",
            "minHeight": "18vw",
            "overflow": "scroll",
            "overflowX": "hidden",
            "width": "100%",
          }
        }
      />
    </div>
  </div>,
  <div
    className="MuiGrid-root"
    style={
      Object {
        "height": "auto",
        "marginBottom": "3vh",
        "marginLeft": "15vw",
        "marginRight": "15vw",
        "marginTop": "2vh",
        "width": "auto",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container"
      style={
        Object {
          "marginBottom": "2.5vh",
        }
      }
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft"
        style={
          Object {
            "marginRight": "2.5vw",
          }
        }
      >
        Categories
      </h4>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "F2994A",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          Add New
        </span>
      </button>
    </div>
  </div>,
  <div
    className="MuiGrid-root"
    style={
      Object {
        "height": "500px",
      }
    }
  />,
  <footer
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "auto",
        "margin": "0",
        "right": "0",
        "textAlign": "center",
        "width": "100vw",
      }
    }
  >
    <div
      style={
        Object {
          "marginTop": "5vh",
        }
      }
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          src="https://advicetracker.life/AdviceLogo.png"
        />
      </a>
    </div>
    <hr
      style={
        Object {
          "width": "auto",
        }
      }
    />
    <p
      className="MuiTypography-root MuiTypography-body2"
      style={
        Object {
          "lineHeight": "24px",
          "margin": "auto",
          "marginTop": "35px",
        }
      }
    >
      © Advice Tracker 2020
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
      style={
        Object {
          "lineHeight": "24px",
          "margin": "auto",
        }
      }
    >
      Made with
       
      <span
        aria-label="Heart"
        role="img"
      >
        ❤️
      </span>
       
      by
       
      <a
        href="https://www.joshbautista.com"
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        Joshua Bautista
      </a>
    </p>
  </footer>,
]
`;
